src/transforms/group.js

```diff
@@ -213,8 +214,13 @@ const reduceDistinct = {
 
 const reduceSum = reduceAccessor(sum);
 
+function sumOrCount(I, V) {
+  if (!("type" in V)) V.type = typeof V.find(v => v != null) === "number";
```

Comment:
You’ll want a WeakMap here instead of mutating V. (I believe that V can sometimes be an externally-provided value, but even if it isn’t, we generally want to avoid mutating inputs.)

Or another simple approach that I believe will work here is single-value memoization: record the last-passed V value and the last-computed type; if the new V value is the same as the old one, return the previously-computed type.

Also, if we do this, it’ll persist across render calls even if the caller mutates the contained values. That might be surprising, but it’s probably rare enough to not worry about.

Reply:
Also, we’ll want an isNumeric helper here modeled after isTemporal and isOrdinal, and we should probably use for-of instead of array.find for consistency:

https://github.com/observablehq/plot/blob/8aafa7c41d08afdaaf8a800a612f9c243d6e9ce9/src/mark.js#L332-L338

Reply:
yes! and memoization will be useful for all of isOrdinal, isTemporal, isNumeric, not just for isNumeric. In a78d7d4 I've centralized the content type detection.